src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -2,3 +2,387 @@
+        server_name=server_name,
+        resource_group_name=resource_group_name).location
+
+
+###############################################
+#                sql db                       #
+###############################################
+
+# Helper class to bundle up database identity properties
+class DatabaseIdentity(object):  # pylint: disable=too-few-public-methods
+    def __init__(self, database_name, server_name, resource_group_name):
+        self.database_name = database_name
+        self.server_name = server_name
+        self.resource_group_name = resource_group_name
+
+
+# Creates a database or datawarehouse. Wrapper function which uses the server location so that
+# the user doesn't need to specify location.
+def _db_dw_create(
+        client,
+        db_id,
+        kwargs):
+
+    # Determine server location
+    kwargs['location'] = get_server_location(
+        server_name=db_id.server_name,
+        resource_group_name=db_id.resource_group_name)
+
+    # Create
+    return client.create_or_update(
+        server_name=db_id.server_name,
+        resource_group_name=db_id.resource_group_name,
+        database_name=db_id.database_name,
+        parameters=kwargs)
+
+
+# Creates a database. Wrapper function which uses the server location so that the user doesn't
+# need to specify location.
+def db_create(
+        client,
+        database_name,
+        server_name,
+        resource_group_name,
+        **kwargs):
+
+    # Verify edition
+    if 'Edition' in kwargs and kwargs['Edition'].lower() == 'datawarehouse':
+        raise CLIError('SQL Data Warehouse can be created with the command `az sql dw create`.')
+
+    return _db_dw_create(
+        client,
+        DatabaseIdentity(database_name, server_name, resource_group_name),
+        kwargs)
+
+
+# Creates a datawarehouse. Wrapper function which uses the server location so that the user doesn't
+# need to specify location.
+def dw_create(
+        client,
+        database_name,
+        server_name,
+        resource_group_name,
+        **kwargs):
+
+    # Set edition
+    kwargs['Edition'] = 'DataWarehouse'
+
+    # Create
+    return _db_dw_create(
+        client,
+        DatabaseIdentity(database_name, server_name, resource_group_name),
+        kwargs)
+
+
+# Common code for special db create modes.
+def _db_create_special(
+        client,
+        source_db,
+        dest_db,
+        kwargs):
+
+    # Determine server location
+    kwargs['location'] = get_server_location(
+        server_name=dest_db.server_name,
+        resource_group_name=dest_db.resource_group_name)
+
+    # Set create mode properties
+    # url parse package has different names in Python 2 and 3.
+    # So we have to import it differently depending on Python version.
+    # pylint: disable=no-name-in-module, import-error
+    import sys
+    if sys.version_info >= (3,):
+        from urllib.parse import quote
+    else:
+        from urllib import quote
```

Comment:
I was hesitant to bring in an entire dependency just for a single line of code. But if you think that's preferable, I can do it.